mock/py/mockbuild/backend.py

```diff
@@ -283,7 +283,8 @@ def build(self, srpm, timeout, check=True, spec=None):
 
             # Check if we will have dynamic BuildRequires, but do not allow it
             hdr = next(util.yieldSrpmHeaders((rebuilt_srpm,)))
-            dynamic_buildreqs = 'rpmlib(DynamicBuildRequires)' in hdr[rpm.RPMTAG_REQUIRES]
+            requires = {util._to_text(req) for req in hdr[rpm.RPMTAG_REQUIRES]}
```

Comment:
Why { (dictionary) ? Shouldn't it be list or tuple?

Reply:
It is a set, not a dictionary

Reply:
And checking if element is in set is faster than in list...

Reply:
ah, ok - this was actually new for me.